### ISSUE TITLE
401 In Git Commit Info Guide

### ISSUE THREAD

**Issue by jeffscottbrown** [micronaut-projects/micronaut-guides#295]
The guide at [guides.micronaut.io/latest/adding-commit-info-gradle-java.html](https://guides.micronaut.io/latest/adding-commit-info-gradle-java.html) provides a link to download the completed example from [guides.micronaut.io/latest/adding-commit-info-gradle-java.zip](https://guides.micronaut.io/latest/adding-commit-info-gradle-java.zip).

When I run that project in a native image built with `./gradlew dBN` I get a 401 when sending a request to the `/info` endpoint.

**Comment by jeffscottbrown** [comment]
The same thing happens when working through the steps in the guide to create the application from scratch.

See the project at [github.com/jeffbrown/micronautguide](https://github.com/jeffbrown/micronautguide). 

The commit history at [github.com/jeffbrown/micronautguide/commits/main](https://github.com/jeffbrown/micronautguide/commits/main) works through the individual steps described in the guide.

Build and run the app:


1. git clone git@github.com:jeffbrown/micronautguide.git
2. cd micronautguide
3. sdk use java 21.1.0.r11-grl
4. ./gradlew nativeImage
5. cd build/native-image
6. ./mn-graalvm-application

Sending a request to `/info` results in a 401.

```
~ $ http :8080/info
HTTP/1.1 401 Unauthorized
connection: keep-alive
date: Fri, 11 Jun 2021 17:40:35 GMT
transfer-encoding: chunked
```

**Comment by jeffscottbrown** [comment]
I have found that if I add a class like the following to the GitHub project that I linked above, requests to the `/info` endpoint return a 200 instead of a 401.

```java
// src/main/java/example/micronaut/SomeClass.java
package example.micronaut;

import javax.inject.Singleton;

@Singleton
public class SomeClass {}

```

The returned JSON document still does not contain the desired commit info.

If I add the following file, then the `/info` endpoint returns the desired commit info:

```
// src/main/resources/META-INF/native-image/resource-config.json

{
  "resources": [
    {"pattern":"\\Qgit.properties\\E"}
  ]
}
```

**Comment by jeffscottbrown** [comment]
The above was tested with the following:

```
$ java -version
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05)
OpenJDK 64-Bit Server VM GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05, mixed mode, sharing)
```

**Comment by jeffscottbrown** [comment]
FYI... @jameskleeh Has suggested that a custom info source bean be added to the sample project so there is at least 1 annotation to process in the app.  That makes part of the problem go away.  I think `resource-config.json` may still be required, at least for Micronaut 2.

**Comment by ilopmar** [comment]
Fixed in #313
